kernel/src/context.rs

```diff
@@ -161,6 +161,9 @@ impl<'a> CurrentUserSpace<'a> {
             check_vaddr(vaddr)?;
         }
 
+        // If `vaddr` is within user space, adjust `max_len` to ensure `vaddr + max_len` does not exceed `MAX_USERSPACE_VADDR`.
+        let max_len = (MAX_USERSPACE_VADDR.checked_sub(vaddr).unwrap_or(max_len)).min(max_len);
```

Comment:
If `checked_sub` fails, it means that `vaddr` is greater than `MAX_USERSPACE_VADDR`. In this case, we shouldn't copy any bytes. So `max_len` should be set to 0.

```rust
let max_len = MAX_USERSPACE_VADDR.checked_sub(vaddr).unwrap_or(0).min(max_len);
```